sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param value Whether or not documentation will be inherited.
+     */
+    public void setInheritDoc(boolean value) {
+        this.inheritDoc = value;
+    }
+
+    /**
+     * @return True if documentation is inherited from the base class, False otherwise.
+     */
+    public boolean getInheritDoc() {
+        return this.inheritDoc;
+    }
+
+    /**
+     * Gets the value of the name decorated with appropriate keywords.
+     *
+     * @return The value of the name decorated with appropriate keywords.
+     */
+    public String getDecoratedName() {
+        StringBuilder decoratedName = new StringBuilder();
+
+        switch (access) {
+            case IMPLICIT:
+            case PACKAGE_PRIVATE:
+                break;
+            case PUBLIC:
+                decoratedName.append("public ");
+                break;
+            case PROTECTED:
+                decoratedName.append("protected ");
+                break;
+            case PRIVATE:
+                decoratedName.append("private ");
+                break;
+            default:
+                logger.logThrowableAsError(new IllegalStateException("Unexpected value: " + access));
+        }
+
+        if (multiplicity == Multiplicity.STATIC) {
+            decoratedName.append("static ");
+        }
+
+        if (this.novelty == Novelty.ABSTRACT) {
+            decoratedName.append("abstract ");
+        }
+
+        if (this.mutability == Mutability.FINAL) {
+            decoratedName.append("final ");
+        }
+
+        if (this.type != null) {
+            decoratedName.append(this.type).append(" ");
+        }
+
+        decoratedName.append(this.name);
+
+        return decoratedName.toString();
+    }
+
+    /**
+     * Add a line of summary text describing the declaration.
+     *
+     * @param text Text for the summary.
+     */
+    public void addSummary(String text) {
+        if (!text.endsWith(".")) {
+            logger.logThrowableAsError(new StyleException("Summary text of declaration '" + this.name + "' must end with a period -- SA1629."));
+        }
+
+        this.summaryLines.add(text);
+    }
+
+    /**
+     * Add a line of remarks text for the declaration.
+     *
+     * @param text Text for the remarks.
+     */
+    public void addRemarks(String text) {
+        if (!text.endsWith(".")) {
+            logger.logThrowableAsError(new StyleException("Remarks text of declaration '" + this.name + "' must end with a period -- SA1629."));
+        }
+
+        this.remarksLines.add(text);
+    }
+
+    /**
+     * Add an attribute to the declaration.
+     *
+     * @param text Attribute text.
+     */
+    public void addAttributes(String text) {
+        this.attributes.add(text);
+    }
+
+    /**
+     * Generate code for the declaration.
+     *
+     * @param codeWriter A {@link CodeWriter} object for generating the declaration code.
+     */
+    public void generateCode(CodeWriter codeWriter) throws IOException {
```

Comment:
No method body. Expected?

Reply:
yes, the subclasses of declaration will override it and they know how to generate the code.

Reply:
look at `JavaMethod` for example.